frontend/packages/console-shared/src/utils/yup-validations.ts

```diff
@@ -1,7 +1,7 @@
 import { TFunction } from 'i18next';
 import * as yup from 'yup';
 
-export const nameRegex = /^[a-z]([a-z0-9]-?)*[a-z0-9]$/;
+export const nameRegex = /^[a-z](?!.*--)[a-z0-9-]*[a-z0-9]$/;
```

Comment:
Please update the unit tests related to this change to make clear which cases doesn't work before and which you want or don't want to allow any more.

The old allowed already 

```
adsf-asdf
asdf-asdf-asdf
```

but not

```
adsf--asdf
asdf--asdf--asdf
```

And this might be one of the issues. Two dashes are allowed!

But your negative lookahead looks like you want to deny that.

Reply:
Two dashes are not allowed on old regex as well but in YAML I am able to create resource with resource name containing two dashes. I will remove that check

---Old regex with two dashes ----
<img width="862" alt="image" src="https://user-images.githubusercontent.com/102503482/229703506-57aa51e8-a9d0-4b5e-8664-45f063f255ad.png">